src/collections/Menu/Menu.js

```diff
@@ -81,10 +81,11 @@ export default class Menu extends Component {
 
     return Children.map(children, (child, i) => {
       const isItem = child.type === MenuItem
+      const isLink = _.has(child, 'props.href') || _.has(child, 'props.link') || _.has(child, 'props.onClick')
 
       if (isItem) {
         const onClick = (e) => {
-          this.handleItemClick(e, i)
+          if (isLink) this.handleItemClick(e, i)
           if (child.props.onClick) child.props.onClick(e, i)
         }
```

Comment:
I don't think we should add the click handler if the user is using subcomponents.  We did this initially elsewhere, but there is just no reliable way to determine which children we should add it to.

React.Children can only loop through the top level children.  If the user uses their own menu item component, then we have no reliable `type` or `displayName` to identify which components are menu items.  They may also use a higher order component or decorator on the MenuItem, in which case the menu item is no longer a top level child but a nested child wrapped in the decorator.

Because of this, I think the only reliable way to pass props to children is when using the `<Menu items={} />` prop.  In this case, we know we can add props to every object in the array.  It is the best way I know of to manage the parent-child coupling.

More here: https://facebook.github.io/react/docs/context.html#parent-child-coupling